src/types/smartSearch.ts

```diff
@@ -161,6 +175,45 @@ export interface SubQueryFilterConfig {
     query_id: number;
 }
 
+interface TaskTimeFrameBefore {
+    before: string;
+}
+
+interface TaskTimeFrameAfter {
+    after: string;
+}
+
+interface TaskTimeFrameBetween {
+    after: string;
+    before: string;
+}
+
+type TaskTimeFrame = boolean
+        | TaskTimeFrameAfter
+        | TaskTimeFrameBefore
+        | TaskTimeFrameBetween;
+
+export interface TaskFilterConfig {
+    campaign?: number;
+    task?: number;
+    type?: 'demographic'
+        | 'offline'
+        | 'share_link'
+        | 'share_image'
+        | 'visit_link';
```

Comment:
There's a type for this union I believe, or rather an enum. Check out `TASK_TYPE` in `src/types/tasks.ts`.

Reply:
fixed